### DISCUSSION TITLE
Handle link taps

### DISCUSSION THREAD

**Discussion by fishcharlie** [gonzalezreal/swift-markdown-ui#241] [Q&A] [answered]
It would be great if there was a way to handle link taps within my own application as opposed to having swift-markdown-ui handle it for me.

This would allow for applications that set a custom browser to work, as well as in app browsers to work as well.

**Comment by gonzalezreal** [comment, marked as answer]
Hi @fishcharlie,

You can set a custom action to handle links in the [`openURL`](https://developer.apple.com/documentation/swiftui/environmentvalues/openurl) environment value.

```swift
Markdown {
  """
  Visit the [terms](https://www.example.com) and [conditions](https://apple.com).
  """
}
.environment(\.openURL, OpenURLAction { url in
  // Modify a state, execute an action, etc., based on 'url'
  return .handled
})
```

The example `Demo` application uses this technique in the `Repository README` section to scroll to the corresponding heading when the user taps an anchor link instead of opening it in Safari:

https://github.com/gonzalezreal/swift-markdown-ui/blob/5f977ba52ddc99a8d891090270b1c5ab1bb9f91d/Examples/Demo/Demo/RepositoryReadmeView.swift#L126-L140
